accepted/0000-yomi-initial-integration.md

```diff
@@ -0,0 +1,363 @@
+
+Internally will still be a Salt state driven by Salt, but we will
+provide a Java API similar to those other mechanisms already in place,
+that will be used to define the pillars that will be applied when a
+new autoinstallation / reinstallation is commanded.
+
+This will imply that the Yomi state is outside the highstate, and that
+this new API needs to be designed.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+Some of the open questions that I am able to identify, are:
+
+## How the minion can find the master automatically.
+
+The current Yomi image (PXE boot and ISO), provides a systemd
+overwrite that is executed before the salt-minion service. This piece
+of code will set a minion ID for the system based on some rules:
+
+- Use the ID injected via the bootloader parameter 'minion='
+
+- If a FQDN is provided via DHCP, this name will be used as the ID
+
+- If no parameter or of the hostname is localhost, the MAC address
+  will be used as minion ID
+
+The question is to validate those rules for the general use case.
+
+## How to find the master node.
+
+In a similar way, an systemd overlay will take care of configuring
+where to find the salt-master node.
+
+- If the bootloader parameter 'master=' is provided, a new file
+  'master.conf' will be added inside '/etc/salt/minion.d'
+
+- If not, the default 'master' name will be used
+
+Is there a way to use DHCP to inject the correct address for the
+salt-master node?
+
+## Disable SSH by default.
+
+The current PXE Boot and ISO image contains a root user than can be
+accessed without password. The problem is that as the SSH service is
+enabled, you can remote login into the new system as root without
+password.
+
+One solution is to disable password-less login via SSH, and inject via
+DHCP some parameters that can enable the login, or inject authorized
+certificates.
+
+## Simplified forms
+
+The prototype provide a set of subforms that cover all the pillar data
+that Yomi can understand. Maybe there is the option of hidding those
+forms and provide a simplified version, based of a single option like:
+(1) what kind of OS want to be installed, (2) simplified partition
+definition.
+
+The list of packages, repositories, service configuration and other
+elements will be described and configured by current parts of Uyuni,
+after the first boot.
+
+## Workflow
+
+In the 'Detailed design' section we pointed a problem with the naive
+workflow, and two alternatives were provided. Which one to choose, and
+how to implement it, is something that needs to be clarified before
+the approval of this RFC.
+
+## Overlap with different pillar and formula
+
+The pillar data for Yomi requires the definition of the language,
+keyboard layout to use, timezone, list of packages, etc. This can
+overlap with other formulas like 'locate', or some other components of
+Uyuni like 'Sofware'
+
+How to resolve this overlap is still an open question.
```

Comment:
We should approach this like YaST modules are doing it. We can re-use the same formulas with forms for installation and later configuration changes, as long as there is logic in the states to determine whether they have to be applied to the changeroot or not.
Software selection is a beast we need to refactor completely in SUSE Manager. For example, we also have package states (which no-one seems to be using) and a field in the activation key form to add packages during activation. For now, a simple form will do.

Reply:
Not sure I understand your last comment. If we're using different naming, we'll obviously have to make it consistent. I'd put it differently: We shouldn't have a separate section on a Yomi form asking for language/timezone/keyboard if there _is_ a form for that already.

Reply:
>  I'd put it differently: We shouldn't have a separate section on a Yomi form asking for language/timezone/keyboard if there is a form for that already.

No question about this.